herd/ASLSem.ml

```diff
@@ -194,16 +222,13 @@ module Make (C : Config) = struct
     let choice (m1 : V.v M.t) (m2 : 'b M.t) (m3 : 'b M.t) : 'b M.t =
       M.bind_ctrl_seq_data m1 (function
         | V.Val (Constant.Concrete (ASLScalar.S_Bool b)) -> if b then m2 else m3
-        | b -> M.bind_ctrl_seq_data (to_int b) (fun v -> M.choiceT v m2 m3))
+        | b ->
+            M.bind_ctrl_seq_data (to_int_signed b) (fun v -> M.choiceT v m2 m3))
```

Comment:
Why signed?

Reply:
Because herd works with signed integers? It should not make any difference here but still

Reply:
OK